bin/check_samplesheet.py

```diff
@@ -83,46 +89,56 @@ def _validate_sample(self, row):
         # Sanitize samples slightly.
         row[self._sample_col] = row[self._sample_col].replace(" ", "_")
 
-    def _validate_first(self, row):
-        """Assert that the first FASTQ entry is non-empty and has the right format."""
-        if len(row[self._first_col]) <= 0:
-            raise AssertionError("At least the first FASTQ file is required.")
-        self._validate_fastq_format(row[self._first_col])
-
-    def _validate_second(self, row):
-        """Assert that the second FASTQ entry has the right format if it exists."""
-        if len(row[self._second_col]) > 0:
-            self._validate_fastq_format(row[self._second_col])
-
-    def _validate_pair(self, row):
-        """Assert that read pairs have the same file extension. Report pair status."""
-        if row[self._first_col] and row[self._second_col]:
-            row[self._single_col] = False
-            first_col_suffix = Path(row[self._first_col]).suffixes[-2:]
-            second_col_suffix = Path(row[self._second_col]).suffixes[-2:]
-            if first_col_suffix != second_col_suffix:
-                raise AssertionError("FASTQ pairs must have the same file extensions.")
-        else:
-            row[self._single_col] = True
-
-    def _validate_fastq_format(self, filename):
-        """Assert that a given filename has one of the expected FASTQ extensions."""
-        if not any(filename.endswith(extension) for extension in self.VALID_FORMATS):
+    def _validate_type(self, row):
+        """Assert that the data type matches expected values."""
+        if not any(row[self._type_col] for datatype in self.VALID_DATATYPES):
             raise AssertionError(
-                f"The FASTQ file has an unrecognized extension: {filename}\n"
-                f"It should be one of: {', '.join(self.VALID_FORMATS)}"
+                f"The datatype is unrecognized: {row[self._type_col]}\n"
+                f"It should be one of: {', '.join(self.VALID_DATATYPES)}"
+            )
+
+    def _validate_data_file(self, row):
+        """Assert that the datafile is non-empty and has the right format."""
+        if len(row[self._file_col]) <= 0:
+            raise AssertionError("Data file is required.")
+        self._validate_data_format(row[self._file_col])
+
+    def _validate_index_file(self, row):
+        """Assert that the indexfile is non-empty and has the right format."""
+        if len(row[self._index_col]) <= 0:
+            raise AssertionError("Data index file is required.")
+        if row[self._file_col].endswith("bam") and not row[self._index_col].endswith("bai"):
+            raise AssertionError("bai index file should be given for bam file.")
+        if row[self._file_col].endswith("cram") and not row[self._index_col].endswith("crai"):
+            raise AssertionError("crai index file shuld be given for cram file.")
+        self._validate_index_format(row[self._index_col])
```

Comment:
This is not necessary because the above two `if` already guarantee that the the index is a `.bai` or a `.crai` (but additionally check that there isn't a mixup like cram+bai or bam+crai)

Reply:
these two conditions are for mixup checking and should work.

Reply:
I mean it's `self._validate_index_format(row[self._index_col])` that's redundant. The two `if` already check the extension

Reply:
removed